packages/shared/lib/clients/sync.client.ts

```diff
@@ -191,51 +189,34 @@ class SyncClient {
                     }
                 });
 
-                await updateSuccessActivityLog(activityLogId as number, false);
+                await updateSuccessActivityLog(activityLogId, false);
 
                 return;
             }
 
-            let handle = null;
-            const jobId = generateWorkflowId(sync, syncName, nangoConnection?.connection_id as string);
+            const jobId = generateWorkflowId(sync, syncName, nangoConnection.connection_id);
 
             if (syncData.auto_start !== false) {
                 if (debug) {
                     await createActivityLogMessage({
                         level: 'debug',
-                        environment_id: nangoConnection?.environment_id as number,
-                        activity_log_id: activityLogId as number,
+                        environment_id: nangoConnection.environment_id,
+                        activity_log_id: activityLogId,
                         timestamp: Date.now(),
-                        content: `Starting sync job ${jobId} for sync ${sync.id}`
+                        content: `Creating sync job ${jobId} for sync ${sync.id}`
                     });
                 }
-                const syncJobId = await createSyncJob(sync.id as string, SyncType.INITIAL, SyncStatus.RUNNING, jobId, nangoConnection);
 
-                if (!syncJobId) {
+                const res = await this.triggerInitialSync({ activityLogId, jobId, nangoConnection, syncId: sync.id!, syncName, debug });
+                if (!res) {
                     return;
```

Comment:
Should we do something here if this call fails?

Reply:
we were not doing anything else before but we can throw yes